generators/app/templates/src/utils/navUtils.js

```diff
@@ -1,33 +1,16 @@
+import React from 'react';
 import { appScreensNavOptions } from '@config/navigation';
 import Routes from '@constants/routes';
 
-export const getCurrentRouteName = navigationState => {
-  if (!navigationState) {
-    return null;
-  }
-  const route = navigationState.routes[navigationState.index];
-  if (route.routes) {
-    return getCurrentRouteName(route);
-  }
-  return route.routeName;
-};
-
-export const getPreviousRouteName = navigationState => {
-  if (!navigationState || !navigationState.index) {
-    return null;
-  }
-  return navigationState.routes[navigationState.index - 1].routeName;
-};
-
-export function inferRoute(screenObj) {
-  const screenName = Routes[Object.keys(screenObj)?.[0]];
-  return (
-    screenName && {
-      [screenName]: {
-        screen: screenObj[screenName],
-        navigationOptions: appScreensNavOptions[screenName],
-        ...screenObj
-      }
-    }
-  );
+export function inferRoute(NavigationStructure) {
+  return function screenComponent(screenObj) {
+    const screenName = Routes[Object.keys(screenObj)?.[0]];
+    return (
+      <NavigationStructure.Screen
+        name={screenName}
+        component={screenObj[screenName]}
+        options={appScreensNavOptions[screenName]}
```

Comment:
if i have a tab as NavigationStructure, it will also use `appScreensNavOptions` for those screens, are we OK with that?

Reply:
Yes, it works with `Tab`, `Drawer` and `Stack`. You will have to understand that in `appScreenNavOptions` you will have to pay attention because the `Tab` screen have different options than the `Stack` screen.

Reply:
And if you are sending a component which is a structure itself, you don't need the options so, it could be undefined.

Reply:
Check this https://reactnavigation.org/docs/screen-options-resolution/